Tests/ExtractFilesTests_Swift.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  ExtractDataTests.m
+//  UnzipKit
+//
+//  Created by Dov Frankel on 7/16/15.
+//  Copyright (c) 2015 Abbey Code. All rights reserved.
+//
+
+import UnzipKit
+
+class ExtractDataTestsSwift: UZKArchiveTestCase {
+
+    func testExtractData_NoPassword() {
+        let testArchives = ["Test Archive (Password).zip"]
+        for testArchiveName in testArchives {
+            do {
+                let archive = try UZKArchive(URL: self.testFileURLs[testArchiveName])
+                let data = try archive.extractDataFromFile("Test File A.txt", progress: { (percentDecompressed) -> Void in
+                    NSLog("Extracting, \(percentDecompressed) complete");
+                })
+            } catch let error as NSError {
```

Comment:
It's your code but imo it's better to handle thing in a swifty way. That approach may be good for objc code but makes no sense in swift code, may result in confusion for a swift programmer.
I copied your own code into swift, though I think it's better to assess different things in swift as it much more error proof in handling nil values than objc